Dockerfile.sles.12.3

```diff
@@ -0,0 +1,66 @@
+#
+# Copyright 2019, Intel Corporation
+#
+# 'recipe' for Docker to build an RPM
+#
+
+# Pull base image
+FROM suse/sles:12.3
+MAINTAINER daos-stack <daos@daos.groups.io>
+
+# use same UID as host and default value of 1000 if not specified
+ARG UID=1000
+
+# Add build user (to keep rpmbuild happy)
+ENV USER build
+RUN useradd -u $UID -ms /bin/bash $USER
+RUN groupadd -g $UID $USER
+
+# Install basic tools
+RUN zypper --non-interactive update
+
+# basic building
+RUN zypper --non-interactive install autoconf automake build                  \
+                                     ca-certificates-mozilla curl createrepo  \
+                                     git libtool lsb-release make             \
+                                     perl-libwww-perl perl-LWP-Protocol-https \
+                                     perl-XML-Parser rpm-build sudo
+
+ENV OPENSUSE_REPO="https://download.opensuse.org/repositories"
+ENV TOOLS_12_3A="openSUSE:/Tools/SLE_12_SP3/openSUSE:Tools.repo"
+ENV TOOLS_12_3="${OPENSUSE_REPO}/${TOOLS_12_3A}"
+ENV HPC_12_3A="science:/HPC:/SLE12SP3_Missing/SLE_12_SP3"
+ENV HPC_12_3="${OPENSUSE_REPO}/${HPC_12_3A}/"
+
+#Temp patch until base docker image rebuilt
+#Disable older SLES repositories.
+RUN zypper --non-interactive mr --disable http-10.8.0.10-099eb6aa; \
+    zypper --non-interactive mr --disable http-10.8.0.10-2a04a526; \
+    zypper --non-interactive mr --disable http-10.8.0.10-33b60288; \
+    zypper --non-interactive mr --disable http-10.8.0.10-4e00da6f; \
+    zypper --non-interactive mr --disable http-10.8.0.10-82d67f5b; \
+    zypper --non-interactive mr --disable http-10.8.0.10-ae069f46; \
+    zypper --non-interactive mr --disable http-10.8.0.10-b2e5365a; \
+    zypper --non-interactive mr --disable http-10.8.0.10-c059c690 || true
+
+RUN zypper --non-interactive ar -f ${TOOLS_12_3}; \
+    zypper --non-interactive ar -f ${HPC_12_3} Science_HPC_SLE_12_SP3; \
+    zypper --non-interactive --gpg-auto-import-keys refresh
+
+# need to run the build command as root, as it needs to chroot
+RUN echo "#includedir /etc/sudoers.d" >> /etc/sudoers; \
+    echo "build ALL=(ALL) NOPASSWD: /usr/bin/build" > /etc/sudoers.d/build
+
+ARG JENKINS_URL=""
+ENV JENKINS_PATH="${JENKINS_URL}job/daos-stack/job"
+ENV JENKINS_ART="lastSuccessfulBuild/artifact/artifacts"
+ARG MNGE_BRANCH="master"
+RUN zypper --non-interactive ar --gpgcheck-allow-unsigned -f \
+    ${JENKINS_PATH}/munge/job/${MNGE_BRANCH}/${JENKINS_ART}/sles12.3/ munge; \
+    zypper --non-interactive ref munge
```

Comment:
For the sake of Dockerfile and docker image consistency, wouldn't it be better to install all of the above repos in the build command rather than in the Docker image directly?  I.e. like it is done with `mock` on EL7.

Doing it this way means not having to do cachebusting repo/package refreshing.

Reply:
Once the Nexus repository is integrated with Jenkins, it will be the only additional repository besides what is built into the base docker images.

The nexus repository groups will take care of everything else.